File: Part2/C++/BasciConcepts/BasicConcepts_Redeclaration.cpp
#include <cstdio>
#include <climits>
#include <cfloat>

int main()
{
    // Do not redeclare variable names in most languages:
    char duplicateCharacter = 'a';
    char duplicateCharacter = 'b';
    
    printf ( "duplicateCharacter: %c\n", duplicateCharacter ) ;
    
    return 0;
}

File: Part2/C#/BaseConcepts/BasicConcepts_Redeclaration.cs
using System;

class Program
{
    static void Main()
    {
        // Do not redeclare variable names in most languages:
        char duplicateCharacter = 'a';
        // This will cause compile error - variable already declared
        char duplicateCharacter = 'b';

        Console.WriteLine("duplicateCharacter: {0}", duplicateCharacter);
    }
}
